packages/components/src/internal/util/utils.ts

```diff
@@ -688,12 +688,21 @@ export function parseCsvString(value: string, delimiter: string, removeQuotes?:
                 end++; // skip double ""
             }
             // if no ending quote, don't remove quotes;
-            if (end === -1) {
-                end = value.indexOf(delimiter, start);
-                if (end === -1) end = value.length;
-                parsedValues.push(value.substring(start, end));
-                start = end + delimiter.length;
-                continue;
+            if (end === -1 || end !== value.length - 1) {
+                let isNextDelimiterOrQuote = true;
```

Comment:
Shouldn't this be initialized to `false`?

Reply:
this is inside the outer `if (ch === '"')` block so it's assumed to be a delimiter. I'll update the variable name to `isCurrentDelimiterOrQuote`